src/main/java/org/broadinstitute/hellbender/utils/Utils.java

```diff
@@ -1041,22 +1048,336 @@ public static boolean xor(final boolean x, final boolean y) {
      * @param reference the reference sequence
      * @param query the query sequence
      */
-    public static int lastIndexOf(final byte[] reference, final byte[] query) {
+    public static int lastIndexOfAtMostTwoMismatches(final byte[] reference, final byte[] query, final int allowedMismatches, int refIndexBound) {
         int queryLength = query.length;
+        if(refIndexBound < 0){
+            refIndexBound = 0;
+        }
 
         // start search from the last possible matching position and search to the left
-        for (int r = reference.length - queryLength; r >= 0; r--) {
-            int q = 0;
-            while (q < queryLength && reference[r+q] == query[q]) {
-                q++;
+        for (int refIndex = reference.length - queryLength; refIndex >= refIndexBound; refIndex--) {
+            int mismatchCount = 0;
+            for (int queryIndex = 0; queryIndex < queryLength && mismatchCount <= allowedMismatches; queryIndex++) {
+                if (reference[refIndex+queryIndex] != query[queryIndex]) {
+                    mismatchCount++;
+                }
             }
-            if (q == queryLength) {
-                return r;
+
+            if (mismatchCount <= allowedMismatches) {
+                return refIndex;
             }
         }
         return -1;
     }
 
+    /**
+     * Finds the location of one indel in the query sequence in relation to the reference sequence
+     * Global Alignment
+     *
+     * Returns the index and size of the indel (as a 2 element int array) or -1 and 0 if an indel less than 4 bases is not found
+     *
```

Comment:
Add a note about tiebreaking here.

Reply:
Specify also that this approach is predicated off of the assumption that the two strings SHOULD be the same length. Also I assume that you account for two indels that happen to be cheaper than one?